### ISSUE TITLE
Additional fixes required for 1.18 compatibility

### ISSUE THREAD

**Issue by BrendanGalloway** [summerwind/actions-runner-controller#128]
Building the lastest version of the controller (including #123) still fail to deploy on a 1.18 cluster:

```$ git fetch summerwind
$ git checkout summerwind/master
HEAD is now at be2e61f Add "alternatives" section to README (#124)
$ kustomize build config/default > latest.yaml
$ # Fix compat with local cert manager
$ sed -i "s/apiVersion: cert-manager\.io\/v1/apiVersion: cert-manager\.io\/v1alpha2/" latest.yaml
$ kubectl apply -f latest.yaml
namespace/actions-runner-system unchanged
customresourcedefinition.apiextensions.k8s.io/horizontalrunnerautoscalers.actions.summerwind.dev configured
customresourcedefinition.apiextensions.k8s.io/runnerdeployments.actions.summerwind.dev configured
customresourcedefinition.apiextensions.k8s.io/runnerreplicasets.actions.summerwind.dev configured
role.rbac.authorization.k8s.io/leader-election-role unchanged
clusterrole.rbac.authorization.k8s.io/manager-role configured
clusterrole.rbac.authorization.k8s.io/proxy-role unchanged
rolebinding.rbac.authorization.k8s.io/leader-election-rolebinding unchanged
clusterrolebinding.rbac.authorization.k8s.io/manager-rolebinding unchanged
clusterrolebinding.rbac.authorization.k8s.io/proxy-rolebinding unchanged
service/controller-manager-metrics-service unchanged
service/webhook-service unchanged
deployment.apps/controller-manager configured
certificate.cert-manager.io/serving-cert unchanged
issuer.cert-manager.io/selfsigned-issuer unchanged
mutatingwebhookconfiguration.admissionregistration.k8s.io/mutating-webhook-configuration configured
validatingwebhookconfiguration.admissionregistration.k8s.io/validating-webhook-configuration configured
The CustomResourceDefinition "runners.actions.summerwind.dev" is invalid:
* spec.validation.openAPIV3Schema.properties[spec].properties[initContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property
* spec.validation.openAPIV3Schema.properties[spec].properties[sidecarContainers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property
* spec.validation.openAPIV3Schema.properties[spec].properties[containers].items.properties[ports].items.properties[protocol].default: Required value: this property is in x-kubernetes-list-map-keys, so it must have a default or be a required property```

```kubectl version
Client Version: version.Info{Major:"1", Minor:"15+", GitVersion:"v1.15.8-beta.0", GitCommit:"6c143d35bb11d74970e7bc0b6c45b6bfdffc0bd4", GitTreeState:"archive", BuildDate:"2020-01-29T00:00:00Z", GoVersion:"go1.14beta1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.8", GitCommit:"9f2892aab98fe339f3bd70e3c470144299398ace", GitTreeState:"clean", BuildDate:"2020-08-13T16:04:18Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}```

**Comment by igorbrigadir** [comment]
I tried this #129 PR: It appears to fix it for me on `1.19` too, with just these warnings:

```
Warning: apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
Warning: admissionregistration.k8s.io/v1beta1 MutatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 MutatingWebhookConfiguration
Warning: admissionregistration.k8s.io/v1beta1 ValidatingWebhookConfiguration is deprecated in v1.16+, unavailable in v1.22+; use admissionregistration.k8s.io/v1 ValidatingWebhookConfiguration
```

Thanks for the fix!

**Comment by BrendanGalloway** [comment]
@igorbrigadir If I understand the chain of dependencies correctly - `apiextensions.k8s.io/v1` includes the fix that allows the defaults to be specified correctly, but is not yet in controller-gen, which is what is used to generate the manifests.  Once that is available, these workarounds can be removed and that warning will also be resolved.
